stream_alert/rule_processor/handler.py

```diff
@@ -56,30 +70,47 @@ def run(self, event):
         Returns:
             [boolean] True if all logs being parsed match a schema
         """
-        LOGGER.debug('Number of Records: %d', len(event.get('Records', [])))
-
-        config = load_config()
+        records = event.get('Records', [])
+        LOGGER.debug('Number of Records: %d', len(records))
+        if not records:
+            return False
+
+        for raw_record in records:
+            # Get the service and entity from the payload. If the service/entity
+            # is not in our config, log and error and go onto the next record
+            service, entity = self.classifier.extract_service_and_entity(raw_record)
+            if not service:
+                LOGGER.error('No valid service found in payload\'s raw record')
+
+            if not entity:
+                LOGGER.error(
+                    'Unable to map entity from payload\'s raw record for service %s',
+                    service)
+
+            if not (service and entity):
+                LOGGER.error('Skipping record: %s', raw_record)
+                continue
 
-        for record in event.get('Records', []):
-            payload = StreamPayload(raw_record=record)
-            classifier = StreamClassifier(config=config)
+            # If the payload's service and entity are found in the config and
+            # contains logs then load the sources for this log
+            if not self.classifier.load_sources(service, entity):
```

Comment:
this statement seems to match the first part of your code comment, but not the `and contains logs` portion?